File: ui/types/request/prompt_eval.ts
export interface CreatePromptEvalRequest {
  prompt_id: number
  system_prompt_input?: Object
  user_prompt_input: Object | string
  name: string
}

export interface UpdatePromptEvalRequest {
  system_prompt_input?: Object
  user_prompt_input?: Object | string
  name?: string
}

File: backend/src/controllers/types/request/prompt_eval.rs
use serde::Deserialize;
use serde_json::Value;

#[derive(Debug, Deserialize)]
pub struct CreateEvalTestRequest {
    pub prompt_id: i64,
    pub system_prompt_input: Option<Value>,
    pub user_prompt_input: Value,
    pub name: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateEvalTestRequest {
    pub system_prompt_input: Option<Value>,
    pub user_prompt_input: String,
    pub name: String,
}
